src/main/java/jep/python/PyPointer.java

```diff
@@ -80,9 +75,9 @@ protected PyPointer(PyObject referrent, MemoryManager memoryManager,
      */
     protected synchronized void dispose() throws JepException {
         if (!disposed) {
+            decref(memoryManager.getThreadState(), pyObject);
             disposed = true;
```

Comment:
Why did you put the `decref` ahead of `disposed = true`?  If decref throws an exception then you could keep attempting to dispose it over and over again, which seems kinda bad.

Reply:
I didn't consider that, I changed it to set the `disosed` before decref.